src/Config.php

```diff
@@ -105,7 +105,59 @@ protected static function defined($key)
             $message = "Aborted trying to redefine constant '$key'. `define('$key', ...)` has already been occurred elsewhere.";
             throw new ConstantAlreadyDefinedException($message);
         }
-        
+
         return false;
     }
+
+    /**
+     * Define set of config items.
+     *
+     * @param array definitions
+     * @return void
+     */
+    protected static function defineSet($definitions)
+    {
+        foreach ($definitions as $const => $def) {
+            self::define($const, $def);
+        }
+    }
```

Comment:
For me "set" means a literal set as in `(kelly, austin) union (roots) == (kelly, austin, roots)`. This api expects a map as in `{kelly: 1, austin: 2, roots: 3}`

```suggestion
    /**
     * Define map of config items.
     *
     * @param array key-value array of definitions
     * @return void
     */
    protected static function defineMap($definitions)
    {
        foreach ($definitions as $const => $def) {
            self::define($const, $def);
        }
    }
```

Reply:
Do we let people pass invalid arrays here? Like what should be do (if anything) if people pass `['WP_ENV']` instead of `['WP_ENV' => 'production']` etc?

Reply:
How about this `@param` to let users, static analyzers (e.g: PHPStan) and IDEs to know we must use string keys?

```diff
-    * @param array key-value array of definitions
+    * @param array<string, mixed> $definitions
```

Reply:
@TangRufus That's a good idea.

I _guess_ we can just not do any input validation and trust the users to use the API correctly. This is what they will run into: https://3v4l.org/DbIEs. We could add a https://stackoverflow.com/a/173479/1585343 to ensure that people don't blow up their config. PHP arrays are pretty silly.